### PR TITLE
Adding new --status flag to bigmler delete

### PR DESCRIPTION
**Pull request by mmerce**

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,15 @@
 History
 -------
 
+3.3.3 (2015-11-04)
+~~~~~~~~~~~~~~~~~~
+
+- Adding --status option to bigmler delete. Selects the resources to delete
+  according to its status (finished if not set). You can check the available
```
Comment (jaor) [inline HISTORY.rst]: nit: their status

File: HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,15 @@
...
+  status in the
```
Comment (jaor) [inline HISTORY.rst]: nit: "status values" (or statuses)

File: bigmler/delete/dispatcher.py [hunk 3]
```diff
@@ -102,8 +115,10 @@ def get_delete_list(args, api, query_list):
             if type_query_list and filter_linked:
                 type_query_list.append(filter_linked)
             if type_query_list:
+                status_code = STATUS_CODES[args.status]
```
Comment (jaor) [inline bigmler/delete/dispatcher.py]: is this fine if `args.status` is not a valid status code name? also, maybe we could let the user specify the status by numeric code? (after all, that's what they see in the API most of the time, no?)

File: bigmler/options/delete.py [hunk 4]
```diff
@@ -191,6 +191,18 @@ def get_delete_options(defaults=None):
...
+
+
+        # Filter the resources to be deleted by its status (finished if
+        # not set)
+        '--status': {
+            'action': 'store',
+            'default': defaults.get('status', "finished"),
+            'choices': ["finished", "faulty", "waiting", "queued", "started",
+                        "in progress", "summarized", "uploading", "unknown",
+                        "runnable"],
```
Comment (jaor) [inline bigmler/options/delete.py]: oh, i see you control here only acceptable names get into args... that's fine, then.  what about also `--status-code` or something similar?

File: bigmler/tests/delete_subcommand_steps.py [hunk 5]
```diff
@@ -110,6 +127,15 @@ def i_delete_source_by_file(step, output_dir=None):
     shell_execute(command, os.path.join(output_dir, "p.csv"), test=None)
 
 
+#@step(r'I delete the source by id using --from-file and --status faulty and the source file storing results in "(.*)"$')
+def i_delete_source_faulty_by_file(step, output_dir=None):
+    if output_dir is None:
+        assert False
```
Comment (jaor) [inline bigmler/tests/delete_subcommand_steps.py]: why not simply `assert output_dir`?

File: bigmler/tests/delete_subcommand_steps.py [hunk 6]
```diff
@@ -170,6 +196,18 @@ def i_create_source_from_file_with_tag(step, data=None, tag=None, output_dir=Non
                " --no-dataset --no-model --store")
     shell_execute(command, os.path.join(output_dir, "p.csv"), test=None)
 
+#@step(r'I create a BigML source from file "(.*)" with tag "(.*)" storing results in "(.*)"')
+def i_create_faulty_source_from_file_with_tag(step, data=None, tag=None, output_dir=None):
+    if data is None or output_dir is None or tag is None:
+        assert False
```
Comment (jaor) [inline bigmler/tests/delete_subcommand_steps.py]: and here: `assert (data and output_dir and tag)`? or, if you want more info about what's failing, 3 asserts:

```python
assert data
assert output_dir
assert tag
```

File: bigmler/tests/delete_subcommand_steps.py [hunk 7]
```diff
@@ -181,6 +219,16 @@ def i_delete_source_newer_and_tag(step, tag=None, output_dir=None):
     shell_execute(command, os.path.join(output_dir, "p.csv"), test=None)
 
 
+
+#@step(r'I delete the source using --newer-than and --status faulty and --source-tag "(.*)" storing results in "(.*)"$')
+def i_delete_source_newer_faulty_and_tag(step, tag=None, output_dir=None):
+    if output_dir is None or tag is None:
+        assert False
```
Comment (jaor) [inline bigmler/tests/delete_subcommand_steps.py]: same as above

File: bigmler/options/delete.py (hunk 4 above)
Comment (mmerce) [inline bigmler/options/delete.py]: i've used the status_code literals in developers docs. In my opinion, they are simpler to remember than numeric codes.

File: bigmler/tests/delete_subcommand_steps.py (hunk 5 above)
Comment (mmerce) [inline bigmler/tests/delete_subcommand_steps.py]: ok

**Pull request merged**